### PR TITLE
Add mso_schema_site_service_graph_node datasource

### PR DESCRIPTION
**Pull request by lhercot**
This PR is a part of the phase-2-resources PR

### CODE REVIEW

File: mso/datasource_mso_schema_site_service_graph_node.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+	msoClient := m.(*client.Client)
+
+	schemaId := d.Get("schema_id").(string)
+	cont, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/schemas/%s", schemaId))
+	if err != nil {
+		return err
+	}
+
+	stateTemplate := d.Get("template_name").(string)
```
Comment (shrsr) [inline mso/datasource_mso_schema_site_service_graph_node.go:91]: Is it viable to query all the templates first to check if the template is available or if there is more than one template with the same name? or is this unnecessary?

File: website/docs/d/mso_schema_site_service_graph_node.html.markdown [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  template_name      = mso_site.example.template_name
+  service_graph_name = "sgtf"
+  service_node_type  = "firewall"
+  service_node_name  = "tfnode2"
+}
+
+```
+
+## Argument Reference ##
+* `schema_id`          - (Required) Schema ID holding Service Graph.
```
Comment (akinross) [inline website/docs/d/mso_schema_site_service_graph_node.html.markdown:27]: the Service Graph?

File: website/docs/d/mso_schema_site_service_graph_node.html.markdown [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+* `template_name`      - (Required) Template Name holding Service Graph.
```
Comment (akinross) [inline website/docs/d/mso_schema_site_service_graph_node.html.markdown:28]: the Service Graph?

File: website/docs/d/mso_schema_site_service_graph_node.html.markdown [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  service_node_type  = "firewall"
+  service_node_name  = "tfnode2"
+}
+
+```
+
+## Argument Reference ##
+* `schema_id`          - (Required) Schema ID holding Service Graph.
+* `template_name`      - (Required) Template Name holding Service Graph. 
+* `service_graph_name` - (Required) Name of Service Graph.
```
Comment (akinross) [inline website/docs/d/mso_schema_site_service_graph_node.html.markdown:29]: the Service Graph?

File: website/docs/d/mso_schema_site_service_graph_node.html.markdown [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+* `service_node_type`  - (Required) Type of Service Node to be attached to this Graph.
```
Comment (akinross) [inline website/docs/d/mso_schema_site_service_graph_node.html.markdown:30]: Type of the Service Node to be attached to the Service Graph?

File: mso/datasource_mso_schema_site_service_graph_node.go (hunk 1 above)
Comment (shrsr) [reply mso/datasource_mso_schema_site_service_graph_node.go:91]: Nvm @akinross pointed me to getTemplateServiceGraphCont that already does it. Resolving this..

**Review by shrsr — APPROVED** [review]
LGTM

File: mso/datasource_mso_schema_site_service_graph_node_test.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccMSOSchemaSiteServiceGraphNode_DataSource(t *testing.T) {
+	resourceName := "mso_schema_site_service_graph_node.test"
+	datasourceName := "data.mso_schema_site_service_graph_node.test"
+	randomParameter := acctest.RandStringFromCharSet(5, "abcdefghijklmnopqrstuvwxyz")
+	randomValue := acctest.RandString(5)
+	name := makeTestVariable(acctest.RandString(5))
```
Comment (sajagana) [inline mso/datasource_mso_schema_site_service_graph_node_test.go:17]: Test failed due to "undefined: makeTestVariable"

File: mso/datasource_mso_schema_site_service_graph_node_test.go [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+					resource.TestCheckResourceAttrPair(resourceName, "service_node_type", datasourceName, "service_node_type"),
+					resource.TestCheckResourceAttr(datasourceName, "service_node_name", "tfnode2"),
+				),
+			},
+		},
+	})
+}
+
+func MSOSchemaSiteServiceGraphNodeDataSourceWithInvalidParentReference(name string) string {
+	resource := CreateServiceNodeResource(name)
```
Comment (sajagana) [inline mso/datasource_mso_schema_site_service_graph_node_test.go:66]: Test failed due to: undefined: CreateServiceNodeResource

File: mso/datasource_mso_schema_site_service_graph_node_test.go [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+		service_graph_name = mso_schema_site_service_graph_node.test.service_graph_name
+		service_node_type  = mso_schema_site_service_graph_node.test.service_node_type
+		service_node_name  = "tfnode2"
+	  }
+	`)
+	return resource
+}
+
+func MSOSchemaSiteServiceGraphNodeDataSourceForRandomAttrName(name, key, value string) string {
+	resource := CreateServiceNodeResource(name)
```
Comment (sajagana) [inline mso/datasource_mso_schema_site_service_graph_node_test.go:80]: Test failed due to: undefined: CreateServiceNodeResource

File: mso/datasource_mso_schema_site_service_graph_node_test.go [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+		service_node_type  = mso_schema_site_service_graph_node.test.service_node_type
+		service_node_name  = "tfnode2"
+		%s				   = "%s"
+	  }
+	`, key, value)
+	return resource
+}
+
+func MSOSchemaSiteServiceGraphNodeDataSourceRequired(name string) string {
+	resource := CreateServiceNodeResource(name)
```
Comment (sajagana) [inline mso/datasource_mso_schema_site_service_graph_node_test.go:95]: Test failed due to: undefined: CreateServiceNodeResource

File: mso/datasource_mso_schema_site_service_graph_node_test.go [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+		service_graph_name = mso_schema_site_service_graph_node.test.service_graph_name
+		service_node_type  = mso_schema_site_service_graph_node.test.service_node_type
+		service_node_name  = "tfnode2"
+	  }
+	`)
+	return resource
+}
+
+func MSOSchemaSiteServiceGraphNodeDataSourceWithoutRequired(name, attr string) string {
+	resource := CreateServiceNodeResource(name)
```
Comment (sajagana) [inline mso/datasource_mso_schema_site_service_graph_node_test.go:109]: Test failed due to: undefined: CreateServiceNodeResource

File: website/docs/d/mso_schema_site_service_graph_node.html.markdown [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+page_title: "MSO: mso_schema_site_service_graph_node"
+sidebar_current: "docs-mso-data-source-schema_site_service_graph_node"
+description: |-
+  Data source for MSO Schema Site Level Service Graph Node
+---
+
+# mso_schema_site_service_graph_node #
+
+Data source for MSO Schema Site Level Service Graph Node.
+
```
Comment (sajagana) [inline website/docs/d/mso_schema_site_service_graph_node.html.markdown:12]: Can we add the GUI Information?

File: website/docs/d/mso_schema_site_service_graph_node.html.markdown [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+## Example Usage ##
+
+```hcl
+data "mso_schema_site_service_graph_node" "test" {
```
Comment (sajagana) [inline website/docs/d/mso_schema_site_service_graph_node.html.markdown:16]: mso_schema_site_service_graph_node - not added in the provider.go file

**Review by sajagana — CHANGES_REQUESTED** [review]

File: website/docs/d/mso_schema_site_service_graph_node.html.markdown [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+* `service_node_type`  - (Required) Type of Service Node to be attached to this Graph.
+* `service_node_name`  - (Required) Name of the Service Graph Node.
+
+
+## Attribute Reference ##
+
+* `site_nodes`             - (Optional) List of maps to provide Site level Node association. This maps should be provided if site is associated with template.
+* `site_nodes.site_id`     - (Optional) Site-Id Attached with the template. Where Service Graph is created. This parameter is required when site is attached with the Template.
+* `site_nodes.tenant_name` - (Optional) Name of Tenant holding the Service Node at site level. This parameter is required when site is attached with the Template.
+* `site_nodes.node_name`   - (Optional) Name of Site level Service Node/Device Name. This parameter is required when site is attached with the Template.
```
Comment (anvitha-jain) [inline website/docs/d/mso_schema_site_service_graph_node.html.markdown:39]: site_nodes.site_id -> site_id

Add indentation for sublevel arguments

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

**Comment by lhercot** [comment]
Closing this PR as we are deprecating mso_schema_site_service_graph_node in preference of mso_schema_site_service_graph resource.

**Pull request closed without merging**